contrib/grpcplugins/action/checkout/main.go

```diff
@@ -113,6 +121,24 @@ func (p *checkoutPlugin) Stream(q *actionplugin.ActionQuery, stream actionplugin
 		}
 	}
 	grpcplugins.Logf(&p.Common, "Checkout completed\n")
+
+	if key != nil {
+		// Install key
+		workDirs, err := grpcplugins.GetWorkerDirectories(ctx, &p.Common)
+		if err != nil {
+			err := fmt.Errorf("unable to get working directory: %v", err)
+			res.Status = sdk.StatusFail
+			res.Details = err.Error()
+			return stream.Send(res)
+		}
+		if err := grpcplugins.InstallSSHKey(ctx, &p.Common, workDirs, sshKey, "", key.Private); err != nil {
```

Comment:
We should install the key where ssh will read it by default:

$HOME/.ssh/id_rsa with 0600

and we have to do `ssh-keyscan -t rsa -p <port> <host> >> $HOME/.ssh/known_hosts`

But $HOME is corrupted by CDS worker, ssh doesn't care about the overrided $HOME. It takes $HOME from /etc/password.


In a CDS v1 script, this works:

```
#!/bin/bash
homedir=$(cat /etc/passwd | grep $(whoami) | cut -d ':' -f 6)
worker key install  proj-robot-cds-fsamin-ssh --file $homedir/.ssh/id_rsa
ssh-keyscan -t rsa -p 7999 git.host.local >> $homedir/.ssh/known_hosts
git clone ssh://git@git.host.local:7999/foo/bar.git
```

Once we install the key in the real $HOME, it's easier.